m3u8/writer.go

```diff
@@ -73,6 +81,24 @@ func (p *MasterPlaylist) Encode() *bytes.Buffer {
 	if p.IndependentSegments() {
 		p.buf.WriteString("#EXT-X-INDEPENDENT-SEGMENTS\n")
 	}
+	if len(p.Defines) > 0 {
+		for _, d := range p.Defines {
+			p.buf.WriteString("#EXT-X-DEFINE:")
+			switch d.Type {
+			case VALUE:
+				p.buf.WriteString("NAME=\"")
```

Comment:
Two things to consider:

1. Use backticks notation to avoid escaping "
2. Introduce functions to write VALUE, QUERYPARAM, and IMPORT that can be used for both master and media playlist and also make the write function more compact.